src/stirling/source_connectors/socket_tracer/protocols/mongodb/stitcher.cc

```diff
@@ -162,6 +162,12 @@ RecordsWithErrorCount<mongodb::Record> StitchFrames(
       resp_frame.consumed = true;
       FlattenSections(&req_frame);
       FlattenSections(&resp_frame);
+
+      // Ignore stitching the request/response if either one is a handshaking frame.
+      if (req_frame.is_handshake || resp_frame.is_handshake) {
+        break;
+      }
```

Comment:
I have a slight preference to set the `consumed` member on these frames to be more explicit about the handling of these frames. I know that our logic still works either way since we clear up stale frames.

Reply:
Do you mean setting the `consumed` field in the request/response frame to `true` again before breaking?

Reply:
Yea.